ToolBox/IBDownloader/Program.cs

```diff
@@ -0,0 +1,151 @@
+                Console.WriteLine("TODATE = yyyymmdd");
+                Environment.Exit(1);
+            }
+            try
+            {
+                // Load settings from command line
+
+                SecurityType securityType;
+                if (!Enum.TryParse(args[0], out securityType)) { throw new ApplicationException("Can't parse securityType "+ args[0]);}
+
+                if (securityType != SecurityType.Equity && securityType != SecurityType.Forex) { throw new ApplicationException("SECURITY_TYPE = eg Equity/Forex"); }
+
+                var tickers = args[1].Split(',');
+                var allResolutions = args[2].ToLower() == "all";
+                var resolution = allResolutions ? Resolution.Second : (Resolution)Enum.Parse(typeof(Resolution), args[2]);
+                var startDate = DateTime.ParseExact(args[3], "yyyyMMdd", CultureInfo.InvariantCulture).ConvertToUtc(TimeZones.NewYork);
+                var endDate = DateTime.ParseExact(args[4], "yyyyMMdd", CultureInfo.InvariantCulture).ConvertToUtc(TimeZones.NewYork);
+
+                
+                //fix end date 
+                endDate = new DateTime(Math.Min(endDate.Ticks, DateTime.Now.AddDays(-1).Ticks));
+
+                // Max number of histoy days
+                int maxDays = 1;
+                if (!allResolutions)
+                {
+                    switch (resolution)
+                    {
+                        case Resolution.Daily:
+                            maxDays = 365;
+                            break;
+                        case Resolution.Hour:
+                            maxDays = 30;
+                            break;
+                        case Resolution.Minute:
+                            maxDays = 10;
+                            break;
+                    }
+                }
+
+                // Load settings from config.json
+                var dataDirectory = Config.Get("data-folder", "../../../Data");
+
+                // Create IB Broker Gateway Runner
+                InteractiveBrokersGatewayRunner.StartFromConfiguration();
+
+                // Create an instance of the downloader
+                const string market = Market.FXCM; // new market for IB ?
+                using (var downloader = new IBDataDownloader())
+                {
+                    foreach (var ticker in tickers)
+                    {
+                        // Download the data
+                        var symbol = Symbol.Create(ticker, securityType, market);
+
+                        var auxEndDate = startDate.AddDays(maxDays); 
+                        auxEndDate = new DateTime(Math.Min(auxEndDate.Ticks, endDate.Ticks));
+
+                        while (startDate < auxEndDate)
+                        {
+                            var data = downloader.Get(symbol, resolution, startDate, auxEndDate);
+                            var bars = data.Cast<QuoteBar>().ToList();
```

Comment:
For equities, `data` is of type `IEnumerable<TradeBar>`, does this line throw?

Reply:
Stefano,  I'm only able to download FOREX. I think this is due to my IB account limitations ( It's just a DEMO account ).  I rather limit the scope of the tool to something I know it's working. Maybe if I sing up for IB I'll come back and add the rest of the assets. 
thanks.